### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Farkal**
Would be great to have same version of `proc-macro2` and `proc-macro-error` for juniper-eager-loading-code-gen (perhaps create workspace ?)

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Are you getting a compile errors because of this? If so I wonder if there is a way to set the version such that we don’t have to keep updating it. At least while the updates are backwards compatible. Would “^x.y” do it? 🤔

I don’t believe the crates belong in the same workspace since they aren’t really related.

**Comment by Farkal** [comment]
No compile error but i have two different version of proc-macro2 because of this package.
After reading [this post]( https://stephencoakley.com/2019/04/24/how-rust-solved-dependency-hell)
I think the best think should be to use x.y (and remove the z) in all your dependencies so rust could find version that match all the requirements easily and remove crates duplicates.

I can update my pull request to use x.y if you want ;)

Edit: The link to the code resolving the dependencies versions -> https://steveklabnik.github.io/semver/src/semver/src/version_req.rs.html#371-423
And `^x.y` is equivalent to `x.y`

File: Cargo.toml [hunk 1]
```diff
@@ -12,14 +12,14 @@ repository = "https://github.com/davidpdrsn/bae.git"
...
 
 [dependencies]
-syn = { version = "1.0.7", features = ["full", "extra-traits"] }
-quote = "1.0.2"
-proc-macro2 = "1.0.6"
-proc-macro-error = "0.3.4"
-heck = "0.3.1"
+syn = { version = "1.0", features = ["full", "extra-traits"] }
+quote = "1.0"
+proc-macro2 = "1.0"
```
Comment (davidpdrsn) [inline Cargo.toml]: I think we should use `"1"` for these three dependencies since minor version bumps should be fine 😊

File: Cargo.toml [hunk 2]
```diff
@@ -12,14 +12,14 @@ repository = "https://github.com/davidpdrsn/bae.git"
...
+proc-macro-error = "0.4"
+heck = "0.3"
 
 [dev_dependencies]
-trybuild = "1.0.17"
+trybuild = "1.0.23"
```
Comment (davidpdrsn) [inline Cargo.toml]: Same here. `"1"` should be fine.

**Comment by davidpdrsn** [comment]
Makes sense. I added some minor comments.

**Comment by Farkal** [comment]
Done ;)

**Comment by davidpdrsn** [comment]
Thanks!

**Comment by davidpdrsn** [comment]
I have shipped version 0.1.3 just now with this patch. I'll get a release of juniper-eager-loading out sometime this week.

**Pull request merged**